pkg/client/jobmanager/jobmanager_client.go

```diff
@@ -0,0 +1,187 @@
+	clientCert, clientKey, err := certs.ClientFactory(userID)
+	if err != nil {
+		return nil, err
+	}
+
+	tc, err := certs.NewClientTransportCredentials(certs.CACert, clientCert, clientKey)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(hostPort, grpc.WithTransportCredentials(tc))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		jm:   jobmanagerv1.NewJobManagerClient(conn),
+		conn: conn,
+	}, nil
+}
+
+// Start invokes an RPC on the JobManager server to start a new job.
+func (c *Client) Start(
+	ctx context.Context,
+	jobName, programPath string,
+	programArgs ...string,
+) (jobID string, err error) {
+
+	job, err := c.jm.Start(ctx, &jobmanagerv1.JobCreationRequest{
+		Name:        jobName,
+		ProgramPath: programPath,
+		Arguments:   programArgs,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return job.Id.Id, nil
+}
+
+// Stop invokes an RPC on the JobManager to stop a job.  If the job with the
+// given jobID isn't running, this operation does nothing.
+func (c *Client) Stop(ctx context.Context, jobID string) error {
+	_, err := c.jm.Stop(ctx, &jobmanagerv1.JobID{Id: jobID})
+
+	return err
+}
+
+// Query invokes an RPC on the JobManager server to retrieve the current status
+// of the job with the given jobID.
+func (c *Client) Query(ctx context.Context, jobID string) (*JobStatus, error) {
+	jobStatus, err := c.jm.Query(ctx, &jobmanagerv1.JobID{Id: jobID})
+	if err != nil {
+		return nil, err
+	}
+
+	return jobStatusRpcToLocal(jobStatus), nil
+}
+
+// Query invokes an RPC on the JobManager server to retrieve the list of jobs
+// started by the user.  If the user is the administrator, then it returns a
+// list of all jobs in the system.
+func (c *Client) List(ctx context.Context) ([]*JobStatus, error) {
+	jobStatusList, err := c.jm.List(ctx, &jobmanagerv1.NilMessage{})
+	if err != nil {
+		return nil, err
+	}
+
+	retList := make([]*JobStatus, len(jobStatusList.JobStatusList))
+
+	for i, js := range jobStatusList.JobStatusList {
+		retList[i] = jobStatusRpcToLocal(js)
+	}
+
+	return retList, nil
+}
+
+// StreamStdout invokes an RPC on the JobManager server to stream the standard
+// output of the job with the given jobID.  This function will block until either
+// (1) the context is interrupted, or (2) the job completes.
+func (c *Client) StreamStdout(ctx context.Context, jobID string, out io.Writer) error {
+	return c.stream(ctx, jobID, out, jobmanagerv1.OutputStream_OutputStream_STDOUT)
+}
+
+// StreamStdout invokes an RPC on the JobManager server to stream the standard
+// error of the job with the given jobID.  This function will block until either
+// (1) the context is interrupted, or (2) the job completes.
+func (c *Client) StreamStderr(ctx context.Context, jobID string, out io.Writer) error {
+	return c.stream(ctx, jobID, out, jobmanagerv1.OutputStream_OutputStream_STDERR)
+}
+
+// Close closes the connection to the JobManager server.
+func (c *Client) Close() {
+	c.conn.Close()
+}
```

Comment:
Why not just make this an `io.Closer`?
```suggestion
func (c *Client) Close() error {
	return c.conn.Close()
}
```